File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/QueryListenerBasic.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.QueryEvent;
import dev.typr.foundations.QueryListener;

@SuppressWarnings("unused")
public class QueryListenerBasic {
    //start
    QueryListener logger = new QueryListener() {
        @Override
        public void beforeQuery(String sql, String name) {
            System.out.println("Executing: " + sql);
        }

        @Override
        public void afterQuery(QueryEvent event) {
            System.out.println(event.name()
                + " completed in " + event.elapsed().toMillis() + "ms");
        }

        @Override
        public void failedQuery(QueryEvent event) {
            System.err.println(event.name()
                + " failed after " + event.elapsed().toMillis() + "ms: "
                + event.error().getMessage());
        }
    };
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/core/QueryListenerBasic.scala
package dev.typr.foundations.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import dev.typr.foundations.{QueryListener, QueryEvent}
import java.sql.Connection

@SuppressWarnings(Array("unused"))
object QueryListenerBasic:
  //start
  object logger extends QueryListener:
    override def beforeQuery(sql: String, name: String): Unit =
      println(s"Executing: $sql")
    override def afterQuery(event: QueryEvent): Unit =
      println(s"${event.name()} completed in ${event.elapsed().toMillis}ms")
    override def failedQuery(event: QueryEvent): Unit =
      System.err.println(s"${event.name()} failed after ${event.elapsed().toMillis}ms: ${event.error().getMessage}")
  //stop
